Feature/rcb-15-db -tructure

SocaciVictor opened the pull request:
I have finished designing the database structure with all the entities.

SocaciVictor committed:
DB-Structured

I created the models for the database structure, including the junction tables.

SocaciVictor committed:
Update DB Structure

I have additionally added the entities: category, difficulty, mealtype, mealtyperecipe.

Ionel14 commented on recipe-box-backend/src/main/java/com/endava/recipebox/model/Recipe.java:
```diff
@@ -0,0 +1,56 @@
+    private String imageUrl;
+
+    @OneToMany(mappedBy = "recipe")
+    private List<MealTypeRecipe> mealType;
```
La relația dintre MealType si Recipe nu puteai avea o relație de tip ManyToMany fără a mai crea acele 2 clase adiacente?

Ionel14 commented on recipe-box-backend/src/main/java/com/endava/recipebox/model/UserIngredient.java:
```diff
@@ -0,0 +1,36 @@
+    @MapsId("ingredientId")
+    @JoinColumn(name = "ingredient_id", nullable = false)
+    private Ingredient ingredient;
+
+    @Column(name = "quantity", nullable = false)
+    private int quantity;
+
+    @Column(name = "unit", nullable = false)
+    private String unit;
```
Aceste 2 campuri sunt identice cu cele din clasa RecipeIngredient, practic este identică cu clasa respectiva, astfel ca nu respecta principiul DRY. In plus quantity este prezent și in Ingredient.

Ionel14 commented:
Si vezi ca nu ai trimis link-ul de la pull request pe git

en-alexandra-costache commented on recipe-box-backend/src/main/java/com/endava/recipebox/model/Recipe.java:
```diff
@@ -0,0 +1,56 @@
+
+   @ManyToOne
+   @JoinColumn(name = "difficulty_id", nullable = false)
+    private Difficulty difficulty;
```
Looks ok, but properties like difficulty or category can also be set as enums (https://thorben-janssen.com/hibernate-enum-mappings/)

SocaciVictor committed:
Change difficulty and add role

I have replaced the Difficulty class with the Difficulty enum and added the Role enum.

SocaciVictor committed:
Replaced class category to enum category

I have replaced the Category class with the Category enum.

SocaciVictor committed:
Replaced class mealtype to enum mealtype

I have replaced the MealType class with the MealType enum.
